libs/Microsoft.MixedReality.WebRTC.Native/src/peer_connection.h

```diff
@@ -398,4 +405,74 @@ class PeerConnection : public TrackedObject {
   PeerConnection(RefPtr<GlobalFactory> global_factory);
 };
 
+/// High level interface for consuming WebRTC audio streams.
+/// The implementation builds on top of the low-level AudioFrame callbacks
+/// and handles all buffering and resampling.
+class AudioReadStream {
+ public:
+  /// Create a new stream which buffers 'bufferMs' milliseconds of audio.
+  /// WebRTC delivers audio at 10ms intervals so pass a multiple of 10.
+  /// Or pass -1 for default buffer size (currently 0.5 seconds).
+  AudioReadStream(PeerConnection* peer, int bufferMs);
+  ~AudioReadStream();
+
+  /// Fill data with samples at the given sampleRate and number of channels.
+  /// If the internal buffer overruns, the oldest data will be dropped.
+  /// If the internal buffer is exhausted, the data is padded with white noise.
```

Comment:
From https://github.com/microsoft/MixedReality-WebRTC/pull/99#issuecomment-567525122 this should be debugging only.

@stephenatwork any reason to pad with noise rather than simply silence here?

Reply:
It was @djee-ms suggestion to help debugging, since there are many states which can lead to silence. It was moderately useful to experimentally choose a buffer size. I suppose it may be more useful to have underrun/overrun status available programmatically.

Reply:
We can have `Read` return the actual consumed samples then.

Also I'd say we can make sine padding opt-in (easier than leave it to higher layers and not too cumbersome).

Reply:
I'd rather have it opt-out. Better have strong indications that something is wrong, and let user ignore them (opt out and get silence instead), than hide a potential issue from the user without them knowing.